packages/firestore/src/core/view.ts

```diff
@@ -323,7 +325,7 @@ export class View {
         oldDocs,
         changes,
         docChanges.mutatedKeys,
-        newSyncState === SyncState.Local,
+        newSyncState === SyncState.Local || fromBundle,
```

Comment:
How do we propagate this state? Let's say I have an active query with Doc A. I then load bundle with Doc B and Doc C, which raises a "fromCache" snapshot. The backend then sends us Doc B and a global snapshot. That snapshot still needs to be fromCache. Do the documents themselves rather than the ViewSnapshot need to carry this state?